custom/dfci_swasth/tests/test_custom_actions.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime, timedelta
+
+from casexml.apps.case.mock import CaseFactory
+from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+from corehq.apps.data_interfaces.tests.test_auto_case_updates import BaseCaseRuleTest
+from corehq.apps.data_interfaces.tests.util import create_empty_rule
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.form_processor.models import CommCareCase
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from custom.dfci_swasth.constants import (
+    CASE_TYPE_PATIENT,
+    CASE_TYPE_CASELOAD,
+    PROP_CCUSER_CASELOAD_CASE_ID,
+    PROP_COUNSELLOR_LOAD,
+    PROP_SCREENING_EXP_DATE,
+    PROP_COUNSELLING_EXP_DATE,
+)
+from custom.dfci_swasth.rules.custom_actions import update_counsellor_load
+from dimagi.utils.parsing import ISO_DATE_FORMAT
+
+
+class UpdateCounsellorLoadTest(BaseCaseRuleTest):
+    domain = 'dfci-swasth'
+
+    def setUp(self):
+        super().setUp()
+        self.domain_obj = create_domain(self.domain)
+
+    def tearDown(self):
+        FormProcessorTestUtils.delete_all_cases()
+        super().tearDown()
+
+    def test_non_patient_case(self):
+        patient_rule = create_empty_rule(
+            self.domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE, case_type=CASE_TYPE_PATIENT,
+        )
+
+        patient_case = CaseFactory(self.domain).create_case(
+            case_type="non-patient",
+        )
+
+        result = update_counsellor_load(patient_case, patient_rule)
+        self.assertEqual(0, result.num_updates)
```

Comment:
this assertion is really not specific here. num_updates could be 0 for different reasons, hence my concern with the implementation of the function update_counsellor_load itself.
If there was specific response that would correspond to unexpected case type, that would be the ideal test here.

Reply:
> specific response

Any suggestion how can we achieve that? 
I have referred existing custom action from https://github.com/dimagi/commcare-hq/blob/bf8245fd2e7ac461ead1ae54737bd8dcec1b3e1d/custom/gcc_sangath/rules/custom_actions.py#L25 to handle no update scenario.

Reply:
can we raise an exception?

Reply:
If we raise an exception from `update_counsellor_load()`, calling action executor should be able to handle it.